io-engine/src/grpc/mod.rs

```diff
@@ -168,6 +173,33 @@ where
         .map_err(|_| Status::resource_exhausted("ENOMEM"))
 }
 
+/// Manage locks accross multiple grpc services.
+pub async fn acquire_subsystem_lock(
+    subsystem: &ResourceSubsystem,
+    resource: Option<String>,
```

Comment:
This would simplify calling side

```suggestion
    resource: Option<&str>,
    ....
    let resource = resource.map(|s| s.to_string());
```

Reply:
With this change, We need to use lifetime for `acquire_subsystem_lock` function for all parameters, as lock_guard lifetime is different than &str lifetime.
With lifetime in place, caller also will have not uniform changes.